### PR TITLE
Allow custom stylesheets to force light/dark mode

### LINKED ISSUE #884: Refactor dark mode to avoid copying and pasting styles
We have a bunch of styles in `ifdb.css` under a `@media (prefers-color-scheme: dark)` media rule.

But many users don't use `prefers-color-scheme`, and instead prefer to manually set dark mode on IFDB. Unfortunately, the only way to do that is to use a custom stylesheet; overwhelmingly, the most popular dark mode stylesheet is Midnight.

That means that when we fix stuff in dark mode, a moderator has to login as MJR and manually edit the Midnight style sheet, copying and pasting the current values from `ifdb.css`.

I can think of a few ways to handle this:

1. We could special case Midnight, and write code to make it automatically use whatever's in the dark mode section of `ifdb.css`
2. We could set a `dark` class on the `<html>` element in JS in the `<head>` if the user prefers dark mode and/or if the user has selected Midnight. Then, `ifdb.css` could stop using a `@media` rule and just check for the presence of the `dark` class, and we could just replace the contents of Midnight with a comment or something.
3. We could add a user preference for dark mode #669 and then switch existing Midnight users to the user preference. (But, is this practically different from the second option, above?)

### PR DESCRIPTION
**Pull request by dfabulich**
Fixes #884, and, kinda, #669 

(Note that testing this requires stopping the Docker instance with `docker compose down`, `./prepare_dev_environment.sh`, and then `docker compose up`.

There's a new setting on custom stylesheets, "Force dark mode," which allows a stylesheet to force IFDB into dark mode or light mode.

When dark mode is forced, the rules that are set to `prefers-color-scheme: dark` are set to `prefers-color-scheme: light, prefers-color-scheme: dark`, i.e. the rule applies no matter what.

When light mode is forced, we remove the rules that are set to `prefers-color-scheme: dark`.

When this merges, I'll login as MJR one last time and set the Midnight CSS to be just `@import('ifdb.css')` with force dark mode. Then, future fixes to `ifdb.css` will automatically apply to Midnight users.

Furthermore, it will then be possible for users to pick a stylesheet that opts out of dark mode, even if their OS is set to prefer dark. I guess I'll create a new "Force Light" stylesheet and mention it in site news or something.

### CODE REVIEW

File: www/styles [hunk 1]
```diff
@@ -30,20 +30,17 @@ if ($setID) {
     if ($userid) {
 
         if ($setID == 'default') {
-            $setVal = 'null';
+            $setVal = null;
```
Comment (salty-horse) [inline www/styles:33]: unused?

File: www/util.php [hunk 2]
```diff
@@ -541,13 +550,25 @@ function showStars($num)
...
     // adding unused class='star-half-checked' and 'star-unchecked' so users can override them in IFDB custom stylesheets
     if (floor($roundedNum) != $roundedNum) {
-        $result .= "<picture><source srcset='/img/dark-images/star-half-checked.svg' media='(prefers-color-scheme: dark)'>"
-            ."<img height=13 class='star-half-checked' src='/img/star-half-checked.svg'></picture>";
+        if ($ssdarkforce === STYLESHEET_DARKFORCE_DARK) {
+            $result .= "<img height=13 class='star-half-checked' src='/img/dark-images/star-half-checked.svg'>";
+        } else if ($ssdarkforce === STYLESHEET_DARKFORCE_LIGHT) {
+            $result .= "<img height=13 class='star-half-checked' src='/img/star-half-checked.svg'>";
+        } else {
+            $result .= "<picture><source srcset='/img/dark-images/star-half-checked.svg' media='(prefers-color-scheme: dark)'>"
```
Comment (salty-horse) [inline www/util.php:558]: can the `media` be on the parent `span` instead of each `source`?

File: www/styles (hunk 1 above)
Comment (dfabulich) [reply www/styles]: fixed

File: www/util.php (hunk 2 above)
Comment (dfabulich) [reply www/util.php:558]: No, `media` is a feature of the `<source>` element. https://developer.mozilla.org/en-US/docs/Web/HTML/Element/source

File: www/styles [hunk 3]
```diff
@@ -185,11 +179,10 @@ if ($setID) {
...
                         // add the row
                         $progress = "SIN003";
-                        $result = mysql_query(
+                        $result = mysqli_execute_query($db,
                             "insert into stylesheets
-                             (title, userid, `desc`, contents, created)
-                             values ('$qTitle', '$userid', '$qDesc',
-                                     '$qContents', now())", $db);
+                             (title, userid, `desc`, contents, dark, created)
+                             values (?, ?, ?, ?, now())", [$formTitle, $userid, $formDesc, $formContents, $formDark]);
```
Comment (salty-horse) [inline www/styles:185]: Missing question mark! 4 instead of 5

Comment (salty-horse) [reply www/styles:185]: Fixed in 62a048dcff1835a30d1e06ed3d0097b32925f76c

**Pull request merged**